File: snail-race-bet/java/src/main/java/snail/race/kata/domain/BetRepositorySimulator.java
package snail.race.kata.domain;

import java.util.ArrayList;
import java.util.List;

public class BetRepositorySimulator implements BetRepository {
    private final List<Bet> bets = new ArrayList<>();

    @Override
    public void register(Bet bet) {
        bets.add(bet);
    }

    @Override
    public List<Bet> findByDateRange(long from, long to) {
        return bets.stream()
                .filter(bet -> bet.timestamp() >= from && bet.timestamp() < to)
                .map(bet -> new Bet(
                        bet.gambler(),
                        new PodiumPronostic(bet.pronostic().first(), bet.pronostic().second(), bet.pronostic().third()),
                        bet.timestamp()
                ))
                .toList();
    }
}

File: snail-race-bet/python/src/domain/bet_repository_simulator.py
from copy import deepcopy
from typing import List

from src.domain.bet import Bet


class BetRepositorySimulator:
    def __init__(self):
        self.in_memory_bets = []

    def register(self, bet: Bet) -> None:
        self.in_memory_bets.append(bet)

    def find_by_date_range(self, from_timestamp: int, to_timestamp: int) -> List['Bet']:
        return [
            deepcopy(bet) for bet in self.in_memory_bets
            if from_timestamp <= bet.timestamp < to_timestamp
        ]